tests/specs/examples/erc20-bin-runtime.k.expected

```diff
@@ -0,0 +1,90 @@
+requires "edsl.md"
+
+module ERC20-BIN-RUNTIME
+    imports public EDSL
+    
+    syntax Bool ::= "unknownBool" [klabel(unknownBool), symbol(), no-evaluators(), function(), functional()]
+    
+    syntax ERC20Contract ::= "ERC20" [klabel(contract_ERC20)]
+    
+    syntax ERC20Function ::= "allowance" "(" Int "," Int ")" 
+    
+    syntax ERC20Function ::= "approve" "(" Int "," Int ")" 
+    
+    syntax ERC20Function ::= "balanceOf" "(" Int ")" 
+    
+    syntax ERC20Function ::= "decimals" "(" ")" 
+    
+    syntax ERC20Function ::= "name" "(" ")" 
+    
+    syntax ERC20Function ::= "symbol" "(" ")" 
+    
+    syntax ERC20Function ::= "totalSupply" "(" ")" 
+    
+    syntax ERC20Function ::= "transfer" "(" Int "," Int ")" 
+    
+    syntax ERC20Function ::= "transferFrom" "(" Int "," Int "," Int ")" 
+    
+    syntax ByteArray ::= ERC20Contract "." ERC20Function [klabel(function_ERC20), macro()]
+    
+    rule  ( ERC20.allowance(OWNER, SPENDER) => #abiCallData("allowance", #address(OWNER), #address(SPENDER)) )
+      requires ( #rangeAddress(OWNER)
+       andBool ( #rangeAddress(SPENDER)
+               ))
```

Comment:
Hmmm, `macro` rules aren't allowed to have `requires` clauses on them actually.

I would actually prefer if we could make these `ensures` clauses, instead of `requires` clauses. Specifically because then we will be adding the needed constraints `#rangeAddress` to those variables for the user, instead of requiring that the user adds them themselves. I'm not sure it's possible or sound though to make `ERC20Contract "." ERC20Function` production a `function` and have the rule read:

```
    rule  ( ERC20.allowance(OWNER, SPENDER) => #abiCallData("allowance", #address(OWNER), #address(SPENDER)) )
      ensures ( #rangeAddress(OWNER)
       andBool ( #rangeAddress(SPENDER)
               ))
```